reactor-core/src/main/java/reactor/core/Exceptions.java

```diff
@@ -725,6 +737,23 @@ public synchronized Throwable fillInStackTrace() {
 		private static final long serialVersionUID = 2491425227432776143L;
 	}
 
+	/**
+	 * A {@link Throwable} that wraps the actual {@code cause} delivered via
+	 * {@link org.reactivestreams.Subscriber#onError(Throwable)} in case of
+	 * {@link org.reactivestreams.Publisher}s that themselves emit items of type
+	 * {@link org.reactivestreams.Publisher}. This wrapper is used to distinguish
+	 * {@code error}s delivered by the upstream sequence from the ones that happen via
+	 * the inner sequence processing chain.
+	 */
+	public static class SourceException extends ReactiveException {
```

Comment:
Intentionally public to allow distinguishing from regular window errors in user code.